src/cow-react/modules/swap/containers/NewSwapModals/index.tsx

```diff
@@ -21,13 +21,19 @@ export const NewSwapModals = React.memo(function (props: NewSwapModalsProps) {
   const { chainId, showNativeWrapModal, showCowSubsidyModal, confirmSwapProps, ethFlowProps } = props
 
   const closeModals = useCloseModals()
-  const history = useHistory()
+  const onCurrencySelection = useOnCurrencySelection()
+  const onImportDismiss = useCallback(
+    (unknownFields: Field[]) => {
+      unknownFields.forEach((field) => onCurrencySelection(field, null))
+    },
+    [onCurrencySelection]
+  )
```

Comment:
I've noticed this written twice. Would be worth it to create a hook for it?

Reply:
Thanks! Fixed